server/src/tenant-datasource/tenant-datasource.service.ts

```diff
@@ -22,21 +22,33 @@ export class TenantDataSourceService {
   public async connectToWorkspaceDataSource(
     workspaceId: string,
   ): Promise<DataSource> {
-    const dataSourceMetadata =
+    const { id, schema } =
       await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
         workspaceId,
       );
+    return this.connectToSchemaDataSource(schema, id);
+  }
 
-    const dataSource = await this.typeormService.connectToDataSource(
-      dataSourceMetadata,
-    );
-
+  /**
+   *
+   * Connect to the schema data source
+   *
+   * @params schema, dataSourceId
+   * @returns
+   */
+  public async connectToSchemaDataSource(
```

Comment:
This could be private

Reply:
it is used in tenant-manager.service so i will let it public